### DISCUSSION TITLE
JOSS paper

### DISCUSSION THREAD

**Discussion by navidcy** [FourierFlows/FourierFlows.jl#280] [General]
I feel that that FourierFlows.jl has matured enough to deserve a small announcement of FourierFlows.jl at, e.g., [JOSS](https://joss.theoj.org).

I suggest that all contributors to be part of this: @apaloczy, @BrodiePearson, and @liasiegelman.

Now I'm not really sure what's the best way forward. FourierFlows.jl builds up the underlying framework/machinery and then other child-packages use this framework to build modules for solving particular PDEs (e.g. [GeophysicalFlows.jl](https://github.com/FourierFlows/GeophysicalFlows.jl), ...) GeophysicalFlows.jl is the package most people use in practice.

Do we
1. write an announcement paper for FourierFlows.jl and another one for GeophysicalFlows.jl?
2. announce FourierFlows as an "ecosystem" and mention the child-packages there?
3. announce GeophysicalFlows.jl and this way better describe its functionalities?

My 2 cents: 1. seems a bit too much. I feel 2. is the option I'd go with... What do others think?

**Comment by BrodiePearson** [comment]
@navidcy I think this sounds like a good idea, thank you for inviting me to be a part of this. Here are my suggestions:

- I would lend the most support to option 1  (separate but complementary announcements), especially given that most JOSS articles are short and concise - so describing the utility of both in a joint article would be a challenge.
- I think the audiences are quite different so it would be hard to hit both audiences with a joint announcement (FourierFlows has a wider breadth of applications especially for developers/code builders, GeophysicalFlows has a more focussed audience with instant applications).
- The FourierFlows announcement should emphasize the breadth of current (and potential future) child-packages, and other child-packages may merit their own announcements when you think it is appropriate.

**Comment by BrodiePearson** [comment]
@navidcy Is Greg also going to be a part of this discussion?

**Comment by navidcy** [comment]
> @navidcy Is Greg also going to be a part of this discussion?

Of course! I presumed @glwagner is watching the repo so he would be notified for the issue. :)

**Comment by glwagner** [comment]
I agree with @BrodiePearson about separate announcements. Since GeophysicalFlows.jl is most often used right now, it could make sense to focus on a JOSS for GeophysicalFlows.jl first.

**Comment by navidcy** [comment]
Sure. But just a remark GeophysicalFlows.jl is mostly used because FourierFlows.jl is just the framework. GeophysicalFlows.jl relied on FourierFlows.jl, so doesn't that makes FourierFlows.jl as equally used? :) 

Perhaps you meant "since GeophysicalFlows.jl seems to be of more interest to many, it could make sense..."?

**Comment by glwagner** [comment]
I agree with what you've said. But the purpose of a JOSS is to 1) advertise software to potential users and 2) provide a citable reference. I think GeophysicalFlows is currently attractive to more users and more likely to need a citable reference when its used in research; therefore if we are going to write a JOSS, we might want to write one about GeophysicalFlows first.

**Comment by navidcy** [comment]
OK @glwagner, phrased this way I agree.

**Comment by navidcy** [comment]
I've added a placeholder for GeophysicalFlows.jl JOSS announcement here:
https://github.com/FourierFlows/GeophysicalFlows.jl/tree/JOSS-paper

@apaloczy, I've included you also since your contribution in FourierFlows.jl is mostly a gfd application of FourierFlows.jl and aligns better, I feel, with GeophysicalFlows.jl.
